src/rest_vol_attr.c

```diff
@@ -479,7 +487,111 @@ RV_attr_open(void *obj, const H5VL_loc_params_t *loc_params, const char *attr_na
 
         /* H5Aopen_by_idx */
         case H5VL_OBJECT_BY_IDX: {
-            FUNC_GOTO_ERROR(H5E_ATTR, H5E_UNSUPPORTED, NULL, "H5Aopen_by_idx is unsupported");
+            htri_t search_ret;
+
+            const char *request_idx_type = NULL;
+
+            switch (loc_params->loc_data.loc_by_idx.idx_type) {
+                case (H5_INDEX_CRT_ORDER):
+                    if (SERVER_VERSION_MATCHES_OR_EXCEEDS(parent->domain->u.file.server_version, 0, 8, 0)) {
+                        request_idx_type = "&CreateOrder=1";
+                    }
+                    else {
+                        FUNC_GOTO_ERROR(
+                            H5E_ATTR, H5E_UNSUPPORTED, NULL,
+                            "indexing by creation order not supported by server versions before 0.8.0");
+                    }
+
+                    break;
+                case (H5_INDEX_NAME):
+                    request_idx_type = "";
+                    break;
+                case (H5_INDEX_N):
+                case (H5_INDEX_UNKNOWN):
+                default:
+                    FUNC_GOTO_ERROR(H5E_ATTR, H5E_CANTALLOC, NULL, "unsupported index type specified");
+                    break;
+            }
+
+            /* Make additional request to server to determine attribute name by index */
+            if (!strcmp(loc_params->loc_data.loc_by_idx.name, ".")) {
+                attribute->u.attribute.parent_obj_type = parent->obj_type;
+                strncpy(attribute->u.attribute.parent_obj_URI, parent->URI, URI_MAX_LENGTH);
```

Comment:
Do we need to keep a separate copy of the parent object's URI rather than just retrieving it on demand?

Reply:
`attribute->u.attribute.parent_obj_URI` is a field on the `RV_attr_t` struct, so that the object an attribute is attached to can be found from the attribute itself at any point (e.g. any time after `RV_attr_open` returns the attribute)